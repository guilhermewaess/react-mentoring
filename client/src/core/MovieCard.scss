.movie-card {
  padding: .5em;

  a {
    color: inherit;
    text-decoration: none;
  }
}
