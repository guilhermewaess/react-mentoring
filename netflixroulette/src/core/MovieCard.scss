.movie-card {
  a {
    color: inherit;
    text-decoration: none;
  }
}
