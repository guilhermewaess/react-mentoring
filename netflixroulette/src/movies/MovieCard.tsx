import * as React from 'react';
import { Card, CardBody, CardImg, CardSubtitle, CardTitle } from 'reactstrap';

export default (props: any) => (
  <Card>
    <CardImg
      top={true}
      width="100%"
      height="400px"
      src={props.movie.poster_path}
      alt="Movie Image"
    />
    <CardBody>
      <CardTitle>{props.movie.title}</CardTitle>
      <CardSubtitle>{props.movie.genres}</CardSubtitle>
    </CardBody>
  </Card>
);
