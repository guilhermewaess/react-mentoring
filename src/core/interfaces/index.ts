import { ActionTypes } from "../store/actions";

export interface IMovie {
  budget: number;
  genres: string[];
  id: number;
  overview: string;
  poster_path: string;
  release_date: string;
  revenue: number;
  runtime: null;
  tagline: string;
  title: string;
  vote_average: number;
  vote_count: number;
}

export interface ICommonState {
  movie: IMovie;
}

export interface ICommonProps {
  movie: IMovie;
}

export interface IAction {
  type: ActionTypes,
  payload: any
}