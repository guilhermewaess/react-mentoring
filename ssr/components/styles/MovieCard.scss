.movie-card {
  padding: 0.5em;

  a {
    color: inherit;
    text-decoration: none;
  }
}
