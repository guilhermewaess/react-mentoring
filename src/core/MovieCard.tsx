import * as React from 'react';
import { NavLink } from 'react-router-dom';
import {
  Badge,
  Card,
  CardBody,
  CardImg,
  CardSubtitle,
  CardTitle,
} from 'reactstrap';
import './MovieCard.scss';

const MovieGenres = ({ genres }: { genres: string[] }) => (
  <div>
    {genres.map((genre, index) => <Badge key={index} color="secondary mr-1">{genre}</Badge>)}
  </div>
);

export default ({ match, movie }: any) => (
  <div className="movie-card">
    <NavLink to={`movie-details/${movie.id}`}>
      <Card>
        <CardImg
          top={true}
          width="100%"
          height="400px"
          src={movie.poster_path}
          alt="Movie Image"
        />
        <CardBody>
          <CardTitle>{movie.title}</CardTitle>
          <CardSubtitle>
            <MovieGenres genres={movie.genres} />
          </CardSubtitle>
        </CardBody>
      </Card>
    </NavLink>
  </div>
);
