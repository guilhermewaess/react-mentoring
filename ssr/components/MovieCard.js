import * as React from 'react';
import {
  Badge,
  Card,
  CardBody,
  CardImg,
  CardSubtitle,
  CardTitle,
} from 'reactstrap';
import './styles/MovieCard.scss';

const MovieGenres = ({ genres }) => (
  <div>
    {genres.map((genre, index) => (
      <Badge key={index} color="secondary mr-1">
        {genre}
      </Badge>
    ))}
  </div>
);

export default ({ movie }) => (
  <div className="movie-card">
    {/* <NavLink to={`/movie-details/${movie.id}`}> */}
    <Card>
      <CardImg
        top={true}
        width="100%"
        height="400px"
        src={movie.poster_path}
        alt="Movie Image"
      />
      <CardBody>
        <CardTitle>{movie.title}</CardTitle>
        <CardSubtitle>
          <MovieGenres genres={movie.genres} />
        </CardSubtitle>
      </CardBody>
    </Card>
    {/* </NavLink> */}
    <style jsx>{`
      .movie-card {
        background-color: black;
      }
    `}</style>
  </div>
);
