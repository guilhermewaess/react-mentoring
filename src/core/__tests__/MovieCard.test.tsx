import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import { IMovie } from '../interfaces';
import MovieCard from './../MovieCard';

describe('MovieCard', () => {
  let component: ShallowWrapper;
  let movie: IMovie;

  const helper = {
    createMovie(): IMovie {
      return {
        budget: 1,
        genres: ['Action', 'Adventure', 'Fantasy', 'Science Fiction'],
        id: 287947,
        overview:
          "We all have a superhero inside us, it just takes a bit of magic to bring it out. In Billy Batson's case, by shouting out one word--SHAZAM!--this streetwise 14-year-old foster kid can turn into the adult superhero Shazam, courtesy of an ancient wizard.Still a kid at heart--inside a ripped, godlike body--Shazam revels in this adult version of himself by doing what any teen would do with superpowers: have fun with them! Can he fly? Does he have X-ray vision? Can he shoot lightning out of his hands? Can he skip his social studies test? Shazam sets out to test the limits of his abilities with the joyful recklessness of a child. But he'll need to master these powers quickly in order to fight the deadly forces of evil controlled by Doctor Thaddeus Sivana.",
        poster_path:
          'https://image.tmdb.org/t/p/w500/yUOJHa9XmB1H0iYodG9Kb3YCc9T.jpg',
        release_date: '2019-04-05',
        revenue: 0,
        runtime: null,
        tagline: '',
        title: 'Shazam!',
        vote_average: 0,
        vote_count: 5,
      };
    },
  };
  beforeEach(() => {
    movie = helper.createMovie();
    component = shallow(<MovieCard movie={movie} />);
  });

  describe('when construct', () => {
    it('should render', () => {
      expect(shallowToJson(component)).toMatchSnapshot();
    });
    it('should render one badge for every genre', () => {
      const genres = component.find('MovieGenres').shallow();
      expect(genres).toMatchSnapshot();
    });
  });
});
