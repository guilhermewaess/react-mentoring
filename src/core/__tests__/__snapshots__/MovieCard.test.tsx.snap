// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard when construct should render 1`] = `
<div
  className="movie-card"
>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="movie-details/287947"
  >
    <Card
      tag="div"
    >
      <CardImg
        alt="Movie Image"
        height="400px"
        src="https://image.tmdb.org/t/p/w500/yUOJHa9XmB1H0iYodG9Kb3YCc9T.jpg"
        tag="img"
        top={true}
        width="100%"
      />
      <CardBody
        tag="div"
      >
        <CardTitle
          tag="h5"
        >
          Shazam!
        </CardTitle>
        <CardSubtitle
          tag="h6"
        >
          <MovieGenres
            genres={
              Array [
                "Action",
                "Adventure",
                "Fantasy",
                "Science Fiction",
              ]
            }
          />
        </CardSubtitle>
      </CardBody>
    </Card>
  </NavLink>
</div>
`;
